### PR TITLE
Apply changes to match the change to a Config struct

### PR DESCRIPTION
**Pull request by medains**
Changes occurred in grafana-api-golang-client - changes here to avoid breaking.

### CODE REVIEW

**Comment by medains** [comment]
This is an illustrative PR.

I think that the changes to grafana/grafana-api-golang-client merit changes to the provider (split auth in the schema to user/pass at least)

Discuss.... ;)

File: grafana/provider.go [hunk 1]
```diff
@@ -41,15 +45,19 @@ func Provider() terraform.ResourceProvider {
...
+	cli := cleanhttp.DefaultClient()
+	cli.Transport = logging.NewTransport("Grafana", cli.Transport)
 	client, err := gapi.New(
-		d.Get("auth").(string),
 		d.Get("url").(string),
+		gapi.Config{
+			BasicAuth: url.UserPassword(splitAuth[0], splitAuth[1]),
+			Client: cli,
+		},
 	)
```
Comment (trotttrotttrott) [inline grafana/provider.go]: This doesn't account for the possibility that `auth` is an API key. What do you think about something like this instead...

```golang
	auth := strings.SplitN(d.Get("auth").(string), ":", 2)
	cli := cleanhttp.DefaultClient()
	cli.Transport = logging.NewTransport("Grafana", cli.Transport)
	cfg := gapi.Config{
		Client: cli,
	}
	if len(auth) == 2 {
		cfg.BasicAuth = url.UserPassword(auth[0], auth[1])
	} else {
		cfg.APIKey = auth[0]
	}
	client, err := gapi.New(d.Get("url").(string), cfg)
```

**Review by trotttrotttrott — COMMENTED** [review]
This is great, thanks for working on it. Just one issue I noticed about using API keys instead of basic auth.

**Review by trotttrotttrott — COMMENTED** [review]
Also, please run `make fmt && go mod vendor` and commit the changes.

**Comment by medains** [comment]
Suggested changes made

**Review by trotttrotttrott — APPROVED** [review]
Excellent. Thanks @medains!

**Pull request merged**